### PR TITLE
Updating documentation

### PR DESCRIPTION
**Pull request by JamesMurkin**
There is likely still plenty to document but this covers some obviously missing features.

### CODE REVIEW

File: docs/helm/server.md [hunk 1]
```diff
@@ -134,3 +134,131 @@ permissionScopeMapping:
...
+  queueLeaseBatchSize: 200
+  minimumResourceToSchedule:
+    memory: 100000000 # 100Mb
+    cpu: 0.25
+  maximalClusterFractionToSchedule:
+    memory: 0.25
+    cpu: 0.25
+```
+
+`queueLeaseBatchSize` This is the number of jobs at the top of a queue the scheduler will look at when trying to find one that'll fit in the available resource.
```
Comment (jankaspar) [inline docs/helm/server.md:192]: I think this is just size of batch, if whole batch is scheduled, we will load additional `queueLeaseBatchSize` amount until we run out of time or global scheduling limit or the queue is empty.

File: docs/helm/server.md [hunk 2]
```diff
@@ -134,3 +134,131 @@ permissionScopeMapping:
...
+```yaml
+scheduling:
+  maximalResourceFractionToSchedulePerQueue:
+    memory: 0.05
+    cpu: 0.05
+  maximalResourceFractionPerQueue:
+    memory: 0.25
+    cpu: 0.25
+```
+
```
Comment (jankaspar) [inline docs/helm/server.md:226]:
```suggestion
All limits are proportional to overall amount of resources in the system. In this example, the queue can use at most 5% of all available cpu and memory.
```

**Review by jimbobby5 — APPROVED** [review]

File: docs/helm/server.md [hunk 3]
```diff
@@ -134,3 +134,131 @@ permissionScopeMapping:
...
+`maximalResourceFractionPerQueue` Is the maximum resource a queue can hold as a percentage of all resource over all clusters.
+
+`maximalResourceFractionToSchedulePerQueue` Is the percentage of total resource a queue can be allocated in a single scheduling round.
+
+Currently scheduling is done in parallel, so it can happen that we exceed the resource limit set in `maximalResourceFractionPerQueue` due to two clusters requesting new jobs at the exact same time.
+
+To migate this, `maximalResourceFractionToSchedulePerQueue` specifies how much can be scheduled in a single round and can be thought of as the margin for error.
```
Comment (jankaspar) [inline docs/helm/server.md:233]:
```suggestion
To mitigate this, `maximalResourceFractionToSchedulePerQueue` specifies how much can be scheduled in a single round and can be thought of as the margin for error.
```

File: docs/helm/executor.md [hunk 4]
```diff
@@ -75,26 +100,113 @@ If the username/password are not present in the server components secret, commun
...
+
+**stuckPodExpiry**
+
+This is how long the executor will let a pod will sit in `Pending` state before it considers the Job stuck.
+
+Once a Job is considered stuck a JobUnableToScheduleEvent will be reported for the job. 
+
+It will then either have:
+ - If the problem is deemed unretryable (for example the image is getting `InvalidImageName`) the job will get a JobFailedEvent and be considered Done
+ - If the probelm is deemed retryable, the job will Have its lease returned to armada-server (JobLeaseReturnedEvent) and the job will be rescheduled
```
Comment (carlocamurri) [inline docs/helm/executor.md:145]:
```suggestion
 - If the problem is deemed retryable, the job will have its lease returned to armada-server (JobLeaseReturnedEvent) and the job will be rescheduled 
```

**Review by jankaspar — DISMISSED** [review]

File: docs/helm/server.md (hunk 1 above)
Comment (jankaspar) [reply docs/helm/server.md:192]: Never mind the effect is actually what is described.

**Review by jankaspar — APPROVED** [review]

**Pull request merged**